Src/ResistanceOperationOverhaul/Classes/UILadderSquadUpgradeScreen.uc

```diff
@@ -1240,6 +1274,90 @@ simulated function UpdateDataSoldierOptions()
 	GetListItem(Index).UpdateDataValue(m_SecondaryWeapon, GetInventoryDisplayText(EquippedItem), OnClickSecondaryWeapon);
 	Index++;
 
+	NumSecondaryAttachmentSlots = 0;
+	if (X2WeaponTemplate(EquippedItem.GetMyTemplate()) != none)
+	{
+		NumSecondaryAttachmentSlots = X2WeaponTemplate(EquippedItem.GetMyTemplate()).NumUpgradeSlots;
+	}
+	
+	Attachments = EquippedItem.GetMyWeaponUpgradeTemplateNames();
+		
+	for (AttachmentIndex = 0; AttachmentIndex < NumSecondaryAttachmentSlots || AttachmentIndex < Attachments.Length; AttachmentIndex++)
+	{
+		if (AttachmentIndex < NumSecondaryAttachmentSlots && Attachments.Length > AttachmentIndex)
+		{
+			AttachmentTemplate = X2WeaponUpgradeTemplate(ItemTemplateManager.FindItemTemplate(Attachments[AttachmentIndex]));
+			if (AttachmentTemplate != none)
+			{
+				GetListItem(Index).EnableNavigation();
+				if (AttachmentIndex >= NumSecondaryAttachmentSlots)
+				{
+					GetListItem(Index).SetDisabled(true, "Cannot change attachment");
+				}
+				GetListItem(Index).UpdateDataValue(m_WeaponAttachment @ string(AttachmentIndex + 1), AttachmentTemplate.GetItemFriendlyNameNoStats(), OnClickSecondaryWeaponAttachment);
+				GetListItem(Index).metadataInt = AttachmentIndex;
+				GetListItem(Index).metadataString = "Attachment";
+				Index++;
+				continue;
+			}
+		}
+
+		GetListItem(Index).EnableNavigation();
+		if (AttachmentIndex >= NumSecondaryAttachmentSlots)
+		{
+			GetListItem(Index).SetDisabled(true, "Cannot change attachment");
+		}
+		GetListItem(Index).UpdateDataValue(m_WeaponAttachment @ string(AttachmentIndex + 1), "None", OnClickSecondaryWeaponAttachment);
+		GetListItem(Index).metadataInt = AttachmentIndex;
+		GetListItem(Index).metadataString = "Attachment";
+		Index++;
+	}
+
+	// Pistol Slot
+	EquippedItem = Soldier.GetItemInSlot(eInvSlot_Pistol, NewGameState, false);
+	GetListItem(Index).EnableNavigation();
+	GetListItem(Index).UpdateDataValue(m_Sidearm, GetInventoryDisplayText(EquippedItem), OnClickSidearm);
```

Comment:
I don't think we want the Pistol Slot to show up if the player is not using a mod that adds it

Reply:
True, Originally I had something set up that would only display it if said mod was installed, but came into some issues with the function actually working half the time (Would display it for the first mission, then forget it existed for some reason.) Probably the way I implemented it, but I suspect a better method would be better in checking for it.